pkg/controllers/provisioning/v1alpha1/reallocation/controller.go

```diff
@@ -87,24 +90,39 @@ func (c *Controller) Reconcile(object controllers.Object) error {
 	}
 
 	// 4. Find Nodes Past TTL with Karpenter Labels
-	expired, err := c.filter.GetExpiredNodes(ctx, provisioner)
+	terminable, err := c.filter.GetTerminableNodes(ctx, provisioner)
 	if err != nil {
-		return fmt.Errorf("getting expired nodes, %w", err)
+		return fmt.Errorf("getting terminable nodes, %w", err)
 	}
-	if len(expired) == 0 {
+	if len(terminable) == 0 {
 		return nil
 	}
 
-	// 5. Cordon each node
-	if err := c.annotator.CordonNodes(ctx, c.filter.GetCordonableNodes(expired)); err != nil {
+	// 5. Cordon each node and label draining
+	if err := c.annotator.CordonNodes(ctx, c.filter.GetCordonableNodes(terminable)); err != nil {
```

Comment:
I'm a little thrown by the differences in patterns in how we're filtering nodes.

Sometimes we get them via filter.GetTerminableNodes(ctx, provisioner), sometimes we get them via c.Filter.GetCordonableNodes(nodes...), c.filter.GetLabeledNodes(ctx, provisioner, map). 

It's not clear to my why this `filter` class behaves in so many different ways. Is there a way that we can collapse these three different modes of operation without sacrificing performance?